### ISSUE TITLE
SAB - Send Invoice Notices - Request option to not send

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1902]
In GitLab by @kdjstudios on Jul 17, 2020, 08:39

**Submitted by:** "Cori Bartlett" <cori.bartlett@answernet.com>

**Helpdesk:** https://www-servicedesk.answernet.com/profiles/ticket/2020-07-14-31363

**Server:** Internal

**Client/Site:** NA

**Account:** NA

**Issue:** 

Both Cori and Cheryl have pointed out that they need a way to 'disable' or 'turn off' these alerts because they have accounts that we do not actually send the invoices to the client. I am confirming with then my proposed solution below will work.

IN summary, the solution I proposed to both of them is to add a "No Invoice" as a media option on the account and then those accounts will be excluded from these "send" invoice functionality. Which will then also in turn remove them from these alerts.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 17, 2020, 08:43

@Robleski @aznbyte 

I also mentioned a "work around" that Cori stated some sites have been doing until we have an actual solution. They are currently setting accounts to the "Pull Bills" and then just clicking the pull bills button but not actually sending the invoices outside of SAB.  

I know we stated we were putting development on hold, but I think this might be a perfect ticket to get Jordan started on and see how he does. If I understand correctly, we really will only be adding one new item in a drop down. Nothing else should need to be updated. We may probably want to jump on a call to further clarify and discuss, but what are both of your thoughts?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 20, 2020, 09:46

Gary wrote:

> Kyle we purposely use the pull bills status to deal with accounts where bills are sent by hand or not at all.  That is as intended.  I don't want a no invoice flag because then someone will invariably use that without realizing that a client is not getting billed

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 21, 2020, 08:20

Gary review today has been canceled this week.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 28, 2020, 09:58

This was reviewed with Gary. We will need to have the following updates done.

1. Remove alerts/notices for only the "pull bills" option. 
2. Update the notice recipients so that the following occurs:
  - NCSM sub billing cycles will only send to the Cori, NCSM, and NCSMBilling emails. Accounting and the Site email will be removed from these alerts.
  - All other billing cycles will have the NCSM and NCSMBilling teams will be removed. 

Please confirm you understand the above requests.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jul 28, 2020, 10:50

@kdjstudios @Robleski 

> Remove alerts/notices for only the "pull bills" option.

We will remove the alert emails for the "pull bills" action for all the sites from both the servers ie. internal and external.

> 1. Update the notice recipients so that the following occurs:

> * NCSM sub billing cycles will only send to the Cori, NCSM, and NCSMBilling emails. Accounting and the Site email will be removed from these alerts.

**For Internal:** Billing cycle action not performed Alerts/emails for all the NCSM billing cycles of all the sites will be sent to only Cori, NCSM, and NCSMBilling emails. And from the recipient's list, we will remove the Accounting and the Site emails.      
**For External:** No NCSM billing cycle on the external, no change required.

> All other billing cycles will have the NCSM and NCSMBilling teams will be removed.   

**For Internal:** Alerts for all the non-NCSM billing cycles of all the sites, billing action not performed alert email will be sent to only Cori, accounting email, and the site email.       
**For External:** No Change

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 28, 2020, 12:49

@ranand 

That looks perfect. Please do this directly on production overnight.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jul 28, 2020, 23:33

@kdjstudios @Robleski 

We have released this on both the production servers.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 29, 2020, 08:33

@ranand 

Thank you for the update. One thing I forgot to clarify or double check is that for all emails we are still BCC the apperrors email


I wrote:

> HI Cori and Team,
> 
> We were able to review this with Gary yesterday and the following updates have been completed per Gary. 
> 
> 1. The Pull Bills option will no longer send notices when a site/billing cycle does not click the button, regardless if there are invoices.
> 2. NCSM sub billing cycle notices will only be sent to the NCSM team. (ncsm@answernet.com and ncsmbilling@answernet.com). While other billing cycles will only be sent to the Site, Cori, and Accounting team.
> 
> Both of the above changes went into effect overnight. We will be closing this ticket. Please let us know if there are any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jul 29, 2020, 22:41

@kdjstudios @Robleski 

> One thing I forgot to clarify or double check is that for all emails we are still BCC the apperrors email

Yes, we are still adding apperrors email to the bcc. We didn't change anything on this front.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 5, 2020, 13:23

So far these appear to be working as expected with the updates. Will wait a week before closing as this is the week sites should have their billing cycle finalized.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 12, 2020, 11:37

closed

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 12, 2020, 11:38

CLosing issue. The Alerts are behaving as expected.
